### DISCUSSION TITLE
creating-an-empty-guid-validation-attribute/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#38] [Announcements]
# creating-an-empty-guid-validation-attribute/

In this post I describe an MVC validation attribute that verifies a GUID doesn't have the default Empty value. I also present a generalised version.

https://andrewlock.net/creating-an-empty-guid-validation-attribute/

**Comment by Jawvig** [comment]
Thanks for this, Andrew, and for confirming this was MIT licenced. Having grabbed the code and followed a few ReSharper refactorings, then adjusted for the fact that our project has `<Nullable>enable</Nullable>`, here is my new version:
```
[AttributeUsage(
    AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter)]
public class NotEmptyGuidAttribute : ValidationAttribute
{
    private const string DefaultErrorMessage = "The {0} field must not be empty";

    public NotEmptyGuidAttribute() : base(DefaultErrorMessage)
    {
    }

    public override bool IsValid(object? value)
    {
        return value switch
        {
            null => true, // NotEmpty doesn't necessarily mean required
            Guid guid => guid != Guid.Empty,
            _ => true
        };
    }
}
```
Note that I also renamed it to `NonEmptyGuidAttribute` as it might otherwise be mistakenly used on a `string` or some other type for which there's an acknowledged 'empty' state.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 16 2019, 06:16, in reply to [Stephen Brouillard](https://disqus.com/by/stephenbrouillard/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/38#discussioncomment-6056338)</em>

---

<p>Agreed, FluentValidator is a great project, for all of the reasons you stated above :) (I did mention it in passing at the end of the post!)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 16 2019, 06:18, in reply to [Rival Abdrakhmanov](https://disqus.com/by/rafaelldi/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/38#discussioncomment-6056350)</em>

---

<p>Quite likely :) As I mentioned in the post, I haven't tested much with the NotDefault attribute - adding a TypeCache for the default values seems like a good optimisation!</p>

[a comment by an automated account is not shown]
